gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -62,11 +71,15 @@ public IdentifyContentsPerExecutor(GCParams gcParams) {
             bloomFilterSize);
   }
 
-  protected Function<String, IdentifiedResult> computeExpiredContentsFunc(
-      Map<String, ContentBloomFilter> liveContentsBloomFilterMap) {
-    return reference -> computeExpiredContents(liveContentsBloomFilterMap, reference);
+  protected SerializableFunction1<scala.collection.Iterator<String>, scala.collection.Iterator<Row>>
```

Comment:
Is it a bit weird that we mix `SerializableFunction1` with java code? Maybe you can make this method static and prevent the need for serialization? (and use `Function`)

Reply:
I tired having all static methods, still gets error for  `interface scala.Function1`
That is why I previously had `SerializableFunction1`

```
Caused by: java.io.NotSerializableException: org.projectnessie.gc.base.IdentifyContentsPerExecutor$$Lambda$2943/0x000000080147d040
Serialization stack:
	- object not serializable (class: org.projectnessie.gc.base.IdentifyContentsPerExecutor$$Lambda$2943/0x000000080147d040, value: org.projectnessie.gc.base.IdentifyContentsPerExecutor$$Lambda$2943/0x000000080147d040@359ae83a)
	- field (class: org.apache.spark.sql.execution.MapPartitionsExec, name: func, type: interface scala.Function1)
```

My branch with this change, if you would like to play around

https://github.com/ajantha-bhat/nessie/commit/aa43d4e72a29a95023e9082de4f57085eb93fd3d